data-prepper-pipeline-parser/src/main/java/org/opensearch/dataprepper/pipeline/parser/transformer/TransformersFactory.java

```diff
@@ -41,35 +53,63 @@ public String getTransformationRulesDirectoryLocation() {
     }
 
     public String getPluginTemplateFileLocation(String pluginName) {
-        if(pluginName == null || pluginName.isEmpty()){
-            throw  new RuntimeException("Transformation plugin not found");
+        if (pluginName == null || pluginName.isEmpty()) {
+            throw new RuntimeException("Transformation plugin not found");
         }
         return templatesDirectoryPath + "/" + pluginName + TEMPLATE_FILE_NAME_PATTERN;
     }
 
-    public String getPluginRuleFileLocation(String pluginName) {
-        if(pluginName == null || pluginName.isEmpty()){
-            throw  new RuntimeException("Transformation plugin not found");
+    public InputStream getPluginTemplateFileStream(String pluginName) {
+        if (pluginName == null || pluginName.isEmpty()) {
+            throw new RuntimeException("Transformation plugin not found");
         }
-        return rulesDirectoryPath + "/" + pluginName + RULE_FILE_NAME_PATTERN;
+        ClassLoader classLoader = TransformersFactory.class.getClassLoader();
+        InputStream filestream = classLoader.getResourceAsStream("templates" + "/" + pluginName + TEMPLATE_FILE_NAME_PATTERN);
+        return filestream;
     }
 
-    public InputStream getPluginRuleFileStream(String pluginName) {
-        if(pluginName == null || pluginName.isEmpty()){
-            throw  new RuntimeException("Transformation plugin not found");
+    public List<Path> getRuleFiles() {
+        // Get the URI of the rules folder
+        URI uri = null;
+        try {
+            uri = getClass().getClassLoader().getResource("rules").toURI();
```

Comment:
Let's use a predefined directory structure to ensure that we are only getting the transform rules:

```
org/opensearch/dataprepper/transforms/
```

Thus, we'd have:

```
org/opensearch/dataprepper/transforms/rules/
org/opensearch/dataprepper/transforms/templates/
```

Reply:
The way i understand this is, we want to move the template and rules to be plugin specific and move this into data-prepper-plugin directory, ie, each plugin gets a rule and template but there are some limitations to this approach. If for example, there is an intent by an user for which plugin does not exist and we want to tranform that to other processors, where do we place the rules and templates in this case. Example for this is with respect to security lake.